src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -54,6 +59,33 @@ public OverseasEntitiesService(OverseasEntitySubmissionsRepository overseasEntit
         this.dateTimeNowSupplier = dateTimeNowSupplier;
     }
 
+    public SubmissionType getSubmissionType(String overseasEntityId) throws ServiceException {
+        if (!isROEUpdateEnabled) {
+            return SubmissionType.REGISTRATION;
+        }
+        Optional<OverseasEntitySubmissionDto> submissionOpt = getOverseasEntitySubmission(
+                overseasEntityId);
+        if (submissionOpt.isEmpty()) {
+            var errorMessage = "Overseas entity submission not found";
```

Comment:
I'd hope an exception already exists for this?

Reply:
It does! Great! It's called `SubmissionNotFoundException`

Reply:
I was viewing this more as a service error  but happy to change to SubmissionNotFound with error message saying that the doc type can't be determined.